src/arch/riscv/pma_checker.cc

```diff
@@ -94,6 +123,61 @@ PMAChecker::takeOverFrom(BasePMAChecker *old)
     PMAChecker* derived_old = dynamic_cast<PMAChecker*>(old);
     assert(derived_old != nullptr);
     uncacheable = derived_old->uncacheable;
+    misaligned = derived_old->misaligned;
+}
+
+Fault
+PMAChecker::checkPAddrAlignment(
+    const RequestPtr &req, BaseMMU::Mode mode, Addr vaddr)
+{
+    Addr addr = 0;
+    if (mode != BaseMMU::Execute) {
+      assert(req->hasPaddr());
+      addr = req->getPaddr();
+      Addr alignSize = mask(req->getArchFlags() & MMU::AlignmentMask) + 1;
+      if (
+          !addressAlign(addr, alignSize) &&
+          !misAlignedSupport(RangeSize(addr, req->getSize()))
+      ) {
+          return createMisalignFault(
+              (req->hasVaddr() ? req->getVaddr() : vaddr), mode);
+      }
+    }
```

Comment:
Sorry I didn't catch this sooner! We're missing spaces here.

Reply:
Hmmm... I have added some comments on the `pma_checker.hh`. Let me know if you have any consideration